crates/polars-python/src/series/aggregation.rs

```diff
@@ -47,124 +49,110 @@ impl PySeries {
     fn mean(&self, py: Python) -> PyResult<PyObject> {
         match self.series.dtype() {
             Boolean => Ok(Wrap(
-                self.series
-                    .cast(&DataType::UInt8)
-                    .unwrap()
-                    .mean_reduce()
+                py.allow_threads(|| self.series.cast(&DataType::UInt8).unwrap().mean_reduce())
```

Comment:
This is a bit nitpicky, so you don't have to change it. But I would just wrap the whole function or match in the gil release handler.